### ISSUE TITLE
Tapping below Bootstrap inputs can focus them in Mobile Safari

### ISSUE THREAD

**Issue by asafbrukarz** [twbs/bootstrap#15968]
This one has had me puzzled for a while and I've been trying to find a more specific cause, but I haven't managed to narrow it down further than Bootstrap.

The fiddle [here](http://jsfiddle.net/9ryem7u4/9/) is set up as follows:
* https://maxcdn.bootstrapcdn.com/bootstrap/3.3.2/css/bootstrap.min.css
* https://maxcdn.bootstrapcdn.com/bootstrap/3.3.2/js/bootstrap.min.js
* Mobile Safari on the iOS simulator (iPhone 6 - iOS 8.1, although I have confirmed this to also be the case on a physical iPhone)

When clicking in the area marked in red:
![ios-simulator-screen-shot-2-mar-2015-11 25](https://cloud.githubusercontent.com/assets/1369002/6433937/eab3f7de-c0cf-11e4-8c5c-1c21043def8e.png)

The input is focused: 
![ios simulator screen shot 2 mar 2015 11 33 45 am](https://cloud.githubusercontent.com/assets/1369002/6433940/0a1ab838-c0d0-11e4-8f0c-27bad7da5263.png)

The issue here is two-fold:
1. Touching anywhere on the screen trigger the input to focus - this even occurs on a scroll event
2. The keyboard does not pop-up. Even if a user wanted to enter text into the input, they need to click it again to trigger the keyboard.

I have been able to confirm the following:
1. Removing **EITHER** bootstrap.min.css or bootstrap.min.js fixes the issue.
2. Removing `class: form-control` from the input style fixes the issue. 
3. Mobile Safari does not attempt to autofocus form elements on its own.

Greatly appreciate your help in tracking down this issue. I'm not sure if it's a browser bug (caused by adding `:focus` styles to an element with the autofocus attribute set), or a javascript event.

**Comment by twbs-lmvtfy** [comment]
Hi @asafbrukarz!

You appear to have posted a live example (http://fiddle.jshell.net/9ryem7u4/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 39, column 5: [E003](https://github.com/twbs/bootlint/wiki/E003): Found one or more `.row`s that were not children of a grid column or descendants of a `.container` or `.container-fluid`

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by asafbrukarz** [comment]
Update the [fiddle](http://jsfiddle.net/9ryem7u4/9/) to remove superfluous rows and grids.

**Comment by cvrebert** [comment]
Possible duplicate of #14532.

**Comment by cvrebert** [comment]
None of our JS should be getting activated in this case, so the bug is very unlikely to be JS-related.

**Comment by asafbrukarz** [comment]
@cvrebert I wouldn't have thought that it was JS related either, but removing the JS file certainly makes a difference.

**Comment by cvrebert** [comment]
Right, but none of our JavaScript acts unbidden on forms or plain text form-controls... Maybe the mere act of adding a document event listener has some weird side-effect on Safari? IDK, just spitballing here.

**Comment by hnrch02** [comment]
Um, according to http://caniuse.com/#feat=autofocus `autofocus` isn't even implemented in iOS Safari.

**Comment by cvrebert** [comment]
Fork without the autofocus attribute: http://jsfiddle.net/gzxtb86c/

**Comment by cvrebert** [comment]
Yup, still happens without autofocus.

**Comment by cvrebert** [comment]
Also, I can't repro the "keyboard doesn't pop up" part with either fiddle. Using iOS 8.1 on an actual iPhone 5S.

**Comment by asafbrukarz** [comment]
You're correct. It's only on the simulator that the keyboard doesn't pop up. 

Also, a touch start event will trigger this behaviour, so it's not limited to tapping.

**Comment by asafbrukarz** [comment]
Sorry, I accidentally closed this issue.

The other point to note is that in your [fork](http://jsfiddle.net/gzxtb86c/) the `focus` style is not applied to the input, where in the original, the blue outline has been applied - even though Mobile Safari should not support `autofocus` at all.

**Comment by asafbrukarz** [comment]
Just confirming that this is still an issue in 3.3.4

**Comment by mdo** [comment]
Is this a browser bug then that needs reporting?

**Comment by asafbrukarz** [comment]
I believe so, but don't have the means to test it. 






On Saturday, 25 Apr 2015 at 6:06 pm, Mark Otto <notifications@github.com>, wrote:

Is this a browser bug then that needs reporting?

**Comment by mdo** [comment]
Anyone willing to test this out? If not, I'm probably going to punt on it until after v4.

**Comment by cvrebert** [comment]
@mdo Already confirmed it earlier.
Filed WebKit bug: https://bugs.webkit.org/show_bug.cgi?id=146244

**Comment by cvrebert** [comment]
Filed Safari bug: http://openradar.appspot.com/21509310

**Comment by cvrebert** [comment]
Fixed in WebKit: http://trac.webkit.org/changeset/191954
